### ISSUE TITLE
gcc49 fails to build on 10.6.8

### ISSUE THREAD

**Issue by axtens** [Homebrew/homebrew-versions#738]
==> Installing gcc49
==> Downloading http://ftpmirror.gnu.org/gcc/gcc-4.9.2/gcc-4.9.2.tar.bz2
######################################################################## 100.0%
==> ../configure --build=x86_64-apple-darwin10.8.0 --prefix=/usr/local/Cellar/gcc49/4.9.2_1 --libdir=/usr/local/Cellar/gcc49/4.9.2_1/lib/gcc/4.9 --enable-languages=c,c++,fortran,java,objc,obj-c++ --pro
==> make bootstrap
../../../libgcc/../config-ml.in: eval: line 160: unexpected EOF while looking for matching `''
../../../libgcc/../config-ml.in: eval: line 161: syntax error: unexpected end of file
make[2]: *** [configure-stage1-target-libgcc] Error 1
make[1]: *** [stage1-bubble] Error 2
make: *** [bootstrap] Error 2

https://gist.github.com/a993ff805209790ec2b4

**Comment by DomT4** [comment]
@mistydemeo If you have some time, would massively appreciate it if you could take a look at this. You've handled some of the most complicated ` gcc ` changes made in Homebrew and will have a better idea that my wild guessing in this situation.

**Comment by mistydemeo** [comment]
It failed when configuring libgcc:

```
config.status: executing default commands
../../../libgcc/../config-ml.in: eval: line 160: unexpected EOF while looking for matching `''
../../../libgcc/../config-ml.in: eval: line 161: syntax error: unexpected end of file
make[2]: *** [configure-stage1-target-libgcc] Error 1
make[1]: *** [stage1-bubble] Error 2
make: *** [bootstrap] Error 2 
```

When I build it locally, I get the same errors from config-ml.in, but it continues regardless.

I'm going to need libgcc's config.log to diagnose further. Could you retry installing gcc49 with the same options and `--debug` and then, when it fails, choose the option to drop to a shell, grab a copy of the `libgcc/config.log` file and upload it to a pastebin? Thanks!

**Comment by mistydemeo** [comment]
Ping.

**Comment by axtens** [comment]
Sorry, on holidays (vacation as you northern-hemispherians call it). But
I'll have a fiddle ... if I can remote in to the Mac successfully.

**Comment by axtens** [comment]
BTW, I tried and succeeded installing gcc48.

**Comment by axtens** [comment]
On Fri, 27 Mar 2015 02:14:33 +0800, Misty De Meo  
<notifications@github.com> wrote:

>
> I'm going to need libgcc's config.log to diagnose further. Could you  
> retry installing gcc49 with the same options and --debug and then, when  
> it >fails, choose the option to drop to a shell, grab a copy of the  
> libgcc/config.log file and upload it to a pastebin? Thanks!
>
>
There is no libgcc subdirectory under  
/private/tmp/gcc4920150401-90306-evgb5g/gcc-4.9.2/build. Closest I get to  
that is libcpp.

Listing directory /private/tmp/gcc4920150401-90306-evgb5g/gcc-4.9.2/build
drwxr-xr-x 18 bruceaxtens wheel 612 Apr 1 20:11 .
drwxr-xr-x 76 bruceaxtens wheel 2584 Apr 1 20:10 ..
-rw-r--r-- 1 bruceaxtens wheel 741014 Apr 1 20:11 Makefile
drwxr-xr-x 5 bruceaxtens wheel 170 Apr 1 20:11  
build-x86_64-apple-darwin10.8.0
-rw-r--r-- 1 bruceaxtens wheel 34089 Apr 1 20:11 config.log
-rwxr-xr-x 1 bruceaxtens wheel 38533 Apr 1 20:11 config.status
drwxr-xr-x 2 bruceaxtens wheel 68 Apr 1 20:11 fixincludes
drwxr-xr-x 636 bruceaxtens wheel 21624 Apr 1 20:22 gcc
drwxr-xr-x 8 bruceaxtens wheel 272 Apr 1 20:11 intl
drwxr-xr-x 37 bruceaxtens wheel 1258 Apr 1 20:12 libbacktrace
drwxr-xr-x 27 bruceaxtens wheel 918 Apr 1 20:12 libcpp
drwxr-xr-x 15 bruceaxtens wheel 510 Apr 1 20:12 libdecnumber
drwxr-xr-x 78 bruceaxtens wheel 2652 Apr 1 20:12 libiberty
-rw-r--r-- 1 bruceaxtens wheel 13 Apr 1 20:11 serdep.tmp
-rw-r--r-- 1 bruceaxtens wheel 7 Apr 1 20:11 stage_current
-rw-r--r-- 1 bruceaxtens wheel 7 Apr 1 20:11 stage_final
-rw-r--r-- 1 bruceaxtens wheel 7 Apr 1 20:11 stage_last
drwxr-xr-x 3 bruceaxtens wheel 102 Apr 1 20:22 x86_64-apple-darwin10.8.0

The config.log in build directory root can be found at  
http://pastebin.com/wGVWT5FJ
The config.log in libcpp can be found at http://pastebin.com/6gXAhfyd
The config.log in gcc can be found at http://pastebin.com/BSswmwyN

**Comment by mistydemeo** [comment]
It this from a separate build? Did it fail in the same way?

All of the provided config.logs succeeded, so unfortunately they're not helpful in diagnosing the problem. :'(

**Comment by benchun** [comment]
I'm seeing something similar on 10.10:

```
[benchun@justright ~]$brew install gcc49 --debug
/usr/local/Library/brew.rb (Formulary::FormulaLoader): loading /usr/local/Library/Taps/homebrew/homebrew-versions/gcc49.rb
==> Installing gcc49 from homebrew/homebrew-versions
/usr/local/Library/brew.rb (Formulary::FormulaLoader): loading /usr/local/Library/Taps/homebrew/homebrew-versions/gmp4.rb
/usr/local/Library/brew.rb (Formulary::FormulaLoader): loading /usr/local/Library/Taps/homebrew/homebrew-versions/libmpc08.rb
/usr/local/Library/brew.rb (Formulary::FormulaLoader): loading /usr/local/Library/Taps/homebrew/homebrew-versions/mpfr2.rb
/usr/local/Library/brew.rb (Formulary::FormulaLoader): loading /usr/local/Library/Taps/homebrew/homebrew-versions/cloog018.rb
/usr/local/Library/brew.rb (Formulary::FormulaLoader): loading /usr/local/Library/Formula/pkg-config.rb
/usr/local/Library/brew.rb (Formulary::FormulaLoader): loading /usr/local/Library/Taps/homebrew/homebrew-versions/isl011.rb
/usr/local/Library/Homebrew/build.rb (Formulary::FromPathLoader): loading /usr/local/Library/Taps/homebrew/homebrew-versions/gcc49.rb
/usr/local/Library/Homebrew/build.rb (Formulary::FormulaLoader): loading /usr/local/Library/Taps/homebrew/homebrew-versions/gmp4.rb
/usr/local/Library/Homebrew/build.rb (Formulary::FormulaLoader): loading /usr/local/Library/Taps/homebrew/homebrew-versions/libmpc08.rb
/usr/local/Library/Homebrew/build.rb (Formulary::FormulaLoader): loading /usr/local/Library/Taps/homebrew/homebrew-versions/mpfr2.rb
/usr/local/Library/Homebrew/build.rb (Formulary::FormulaLoader): loading /usr/local/Library/Taps/homebrew/homebrew-versions/cloog018.rb
/usr/local/Library/Homebrew/build.rb (Formulary::FormulaLoader): loading /usr/local/Library/Taps/homebrew/homebrew-versions/isl011.rb
/usr/local/Library/Homebrew/build.rb (Formulary::FormulaLoader): loading /usr/local/Library/Formula/pkg-config.rb
==> Downloading http://ftpmirror.gnu.org/gcc/gcc-4.9.2/gcc-4.9.2.tar.bz2
==> Downloading from: http://gnu.mirror.iweb.com/gcc/gcc-4.9.2/gcc-4.9.2.tar.bz2
######################################################################## 100.0%
==> ../configure --build=x86_64-apple-darwin14.3.0 --prefix=/usr/local/Cellar/gcc4
==> make bootstrap
make[3]: *** [graphite.o] Error 1
rm gcc.pod
make[2]: *** [all-stage1-gcc] Error 2
make[1]: *** [stage1-bubble] Error 2
make: *** [bootstrap] Error 2
/usr/local/Library/Homebrew/debrew.rb:11:in `raise'
BuildError: Failed executing: make bootstrap
1. raise
2. ignore
3. backtrace
4. irb
5. shell
Choose an action: 5
When you exit this shell, you will return to the menu.
bash-3.2$ ls
Makefile                        libcpp
build-x86_64-apple-darwin14.3.0 libdecnumber
config.log                      libiberty
config.status                   serdep.tmp
fixincludes                     stage_current
gcc                             stage_final
intl                            stage_last
libbacktrace                    x86_64-apple-darwin14.3.0
bash-3.2$ pwd
/private/tmp/gcc4920150618-67226-a4kb2z/gcc-4.9.2/build
```

@mistydemeo, is there any other info that would be useful? I believe I've had this work fine on other machines, and I can't discern any difference with this one... its `brew update` and `brew doctor` are normal. 

@axtens, did you ever resolve your issue?

**Comment by DomT4** [comment]
`brew gist-logs gcc49` will be useful.

**Comment by benchun** [comment]
@DomT4 that is a pretty cool feature. Here's what came out: https://gist.github.com/anonymous/3fdc85437ac0a5f64d4b

**Comment by whitej125** [comment]
Also running into this myself. Upgrading to GCC5 borked R.  Now trying to reinstall gcc49... getting a similar fault with `brew install gcc49 --with-fortran`.

**Comment by whitej125** [comment]
It looks like there were some upstream fixes applied to 4.9.1 back when 4.9.1 was the latest gcc version available.  https://github.com/Homebrew/linuxbrew/issues/154

**Comment by tnilanon** [comment]
Ran into this same issue.

The problem, it turns out, is that `bash` parses `ac_configure_args` incorrectly, exactly the issue @whitej125 was referring to.

To elaborate, `bash` does not respected single-quoted string by default. It will cut
`'--with-pkgversion=Homebrew gcc49 4.9.3 --without-multilib'`
to
`'--with-pkgversion=Homebrew`, `gcc49`, `4.9.3`, and `--without-multilib'`
creating unmatched quotes in the process.

It was **reported** upstream here: https://gcc.gnu.org/ml/gcc-patches/2014-03/msg00718.html and **fixed** here: https://github.com/gcc-mirror/gcc/commit/52fd2e17c7426500a4e47047f6a1ea9d2a889db3.diff

**Comment by jedemenge-it** [comment]
Hi,

i have the same error (config-ml.in) with gcc48 on 10.10. Seems that this patch is not included in the upstream version, or am i getting this wrong?

@tnilanon Would you please tell me how i can apply this patch within homebrew? 

Best 
Andreas

**Comment by tnilanon** [comment]
Hey,

Yes. The patch is **not** included in `gcc 4.9.3`, so I would assume that it's not in `4.8` either.

I did the patch, but I still couldn't build `gcc`. Different error though. As I am on a deadline, I decided to use `gcc` on `ubuntu` on a virtual machine instead.

As you are building `gcc 4.8`, I don't think my patch will work with yours. Basically you just need to append `__END__` and the `diff` (my personal choice is `git diff`) to the end of `/usr/local/Library/Taps/homebrew/homebrew-versions/gcc48.rb` and add `patch :DATA` before `def install`. More info here https://github.com/Homebrew/homebrew/blob/master/share/doc/homebrew/Formula-Cookbook.md#patches

Will update when I finally have some time to figure this out.
